### PR TITLE
Add MKL documentation

### PR DESCRIPTION
**Pull request by ykim362**
## Description ##
Add a documentation for MKL integration

### Essentials ###
- [x] Passed code style checking (`make lint`)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [x] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [x] docs --> How to --> build_mxnet_with_intelmkl.md

### CODE REVIEW

**Comment by szha** [comment]
What happens with [MKL_README.md](https://github.com/apache/incubator-mxnet/blob/master/MKL_README.md)?

File: docs/how_to/build_mxnet_with_intelmkl.md [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+tools and libraries on a variety of platforms. This tutorial provides one 
+such recipe describing steps to build and install MXNet with Intel MKL on 
+an Ubuntu-based system, version 16.04.
+
+## Installing
+
+1. Update your operating system and install dependencies:
+
+   ```bash
+    $ sudo apt-get update && apt-get install -y build-essential git libopencv-dev curl gcc libatlas-base-dev python python-pip python-dev python-opencv graphviz python-scipy python-sklearn
```
Comment (szha) [inline docs/how_to/build_mxnet_with_intelmkl.md]: All other build instructions are in https://github.com/apache/incubator-mxnet/tree/master/docs/install, so please incorporate these instructions there.

**Comment by ykim362** [comment]
This is an effort to add MKL instructions into the official documentations. It has a little different purpose from readme file.

File: docs/how_to/build_mxnet_with_intelmkl.md [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+   ```
+
+2. Delete any old or outdated versions of Intel MKL from `/tmp` before 
+   downloading and copying the newest version to `/usr/local`:
+
+   ```bash
+   $ cd /tmp && rm -rf mklml_lnx_*   
+   $ curl -L https://github.com/01org/mkl-dnn/releases/download/v0.10/mklml_lnx_2018.0.20170908.tgz | tar xz
```
Comment (szha) [inline docs/how_to/build_mxnet_with_intelmkl.md]: This seems outdated given that your PR #9218 is already merged.

File: docs/how_to/build_mxnet_with_intelmkl.md [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+  
+```
+export OMP_NUM_THREADS=$(($(grep 'core id' /proc/cpuinfo | sort -u | wc -l)*2))
+```
+   
+   
+#### For Intel® Xeon Phi™ Processors
+   
+```
+export OMP_NUM_THREADS=$(($(grep 'core id' /proc/cpuinfo | sort -u | wc -l)))export KMP_AFFINITY=granularity=fine,compact,1,0
```
Comment (szha) [inline docs/how_to/build_mxnet_with_intelmkl.md]: This looks like it belongs in [perf.md](https://github.com/apache/incubator-mxnet/blob/master/docs/faq/perf.md)

File: docs/how_to/build_mxnet_with_intelmkl.md (hunk 2 above)
Comment (ykim362) [reply docs/how_to/build_mxnet_with_intelmkl.md]: I agree. It should have been changed.

File: docs/how_to/build_mxnet_with_intelmkl.md (hunk 3 above)
Comment (ykim362) [reply docs/how_to/build_mxnet_with_intelmkl.md]: I see. I will revise it.

**Comment by szha** [comment]
@ykim362 not sure if I follow. Given that both MKL_README.md and the added content in your PR have build instructions, should MKL_README.md be deleted?

Also, MKL is one of many other build options available in mxnet. Thus, I'd suggest that the corresponding build instructions be provided in the existing "build from source" instructions, so that it's easier for the users to find and follow.

File: docs/how_to/build_mxnet_with_intelmkl.md (hunk 1 above)
Comment (ykim362) [reply docs/how_to/build_mxnet_with_intelmkl.md]: I got it. I will have an internal discussion regarding this.

**Comment by ykim362** [comment]
@szha I got your point. I will have an internal discussion and make a new PR to address your concerns.

**Pull request closed without merging**